ooniprobe/AppDelegate.mm

```diff
@@ -143,6 +155,53 @@ -(void)application:(UIApplication *)application didReceiveLocalNotification:(UIL
     //TODO start your long running bg task here
 }
 
+- (BOOL)application:(UIApplication *)application handleOpenURL:(NSURL *)url {
```

Comment:
@lorenzoPrimi my understanding is that `handleOpenUrl` is an hardcoded name for iOS. This makes it a little more complex to read the code &mdash; compared to the case where the callback name was more specific &mdash; since I do not immediately understand who calls this function. I believe it would be helpful to have a short documentation (one or two lines) before a function (especially for this fixed-name callbacks) to help the reader :-).

Reply:
Ah, uhm, okay: now I figured that `handleOpenURL` is the method called by iOS on the `UIApplication` when the user clicks on a URL. It was honestly much less difficult than I initially thought (_oops_...). Anyway, maybe adding one line of documentation may be a good idea anyway.

Reply:
_Question_: how do you register handling URLs with iOS? I assume with the entitlements change. If this is correct, I think I have missed the place where `ooni://...` URLs are registered.

If my additional guess that here you will handle both `ooni://...` and `https://run.ooni.io/...` links, then I believe the code should be updated to make sure it handles both cases.

Besides, maybe, checking at the beginning that the URL either starts with `ooni://...` or `https://run.ooni.io/...` _may_ be useful in future to detect possible errors that we could make where we accidentally pass to this function _other_ URLs.

Reply:
Everything is the Info.plist file.
The links on http are handled by this other method

- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity  restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler{

Reply:
Okay, maybe can you add comments saying that? It simplifies reading the code.